boards/arm/arty/Kconfig.defconfig

```diff
@@ -21,14 +21,4 @@ config NUM_IRQS
 config GPIO
 	default y if "$(dt_nodelabel_enabled,daplink_qspi_mux)"
 
-if FLASH
-
-config SPI
-	default y
-
-config SPI_NOR
-	default y if SPI_XLNX_AXI_QUADSPI
```

Comment:
Is removal of this correct? y here depends on another symbol.

Reply:
It is required to have `SPI_NOR` select `SPI`, otherwise a dependency loop exists:
`SPI_XLNX_AXI_QUADSPI depends on SPI -> SPI_NOR selects SPI -> SPI_NOR depends on SPI_XLNX_AXI_QUADSPI`.

I must admit the change for `SPI_NOR` went a bit over my head, maybe I should only have `SPI_FLASH_AT45` in this PR (which is what I am working with).

Reply:
I am fine with the SPI change, in flash drivers, but I am not sure how to address board configurations.
Board configurations  may still want to select `SPI_NOR`, to enable NOR flash support by default, when FLASH is enabled.

The `default y if SPI_XLNX_AXI_QUADSPI`  can be false if somebody disables it and `SPI_NOR`, here, may depend on `SPI_XLNX_AXI_QUADSPI` because this might be the SPI driver, the SPI_NOR hangs on.

Anyway, this requires some comments from board maintainers.

Reply:
The cases where I removed the board default for `SPI_NOR` were only for `if FLASH ... endif` I think.